hq/app/aws/ec2/EC2.scala

```diff
@@ -55,11 +55,15 @@ object EC2 {
     for {
       sgResult <- TrustedAdvisorSGOpenPorts.getSGOpenPorts(supportClient)
       sgUsage <- getSgsUsage(sgResult, account)
-      flaggedSgs <- updateSgsWithTags(account, sgResult.flaggedResources.filter(_.status != "ok")){ (ec2Client, request) =>
-        handleAWSErrs(awsToScala(ec2Client.describeSecurityGroupsAsync)(request))
-      }
-      vpcs <- getVpcs(account, flaggedSgs)(getVpcsDetails)
-      flaggedSgsWithVpc = addVpcName(flaggedSgs, vpcs)
+      flaggedSgs = sgResult.flaggedResources.filter(_.status != "ok")
```

Comment:
This is ok for now but the for comp is a little on the long side. There's tension between the positive impacts of "be explicit in the for comp and avoid deeply nested function calls" and "keep the for comprehension as simple as possible because it's hard to test".

While there's no "correct" answer, there are a couple of areas where we have sequential "pure" computations (`=` rather than `<-`) where the intermediate result is not used elsewhere. This is a sign that we're doing logic in the for comp. For example:

```scala
sgTagDetails = describeSecurityGroupsResults.flatMap(extractTagsForSecurityGroups).toMap
enrichedFlaggedSgs = enrichSecurityGroups(flaggedSgs, sgTagDetails)
```

I think we could probably refactor this to do all the IO up front and then combine some of the pure work we do into a function that we can test. Let's look at doing this afterwards though, particularly as we're aware the info might be coming from different locations (cached data) soon.